packages/@react-native-windows/cli/src/runWindows/utils/autolink.ts

```diff
@@ -591,6 +592,155 @@ async function updateAutoLink(
   }
 }
 
+export function ensureWinUIDialect(
+  slnFile: string,
+  windowsProjectConfig: WindowsProjectConfig,
+  checkMode: boolean,
+) {
+  const buildFlagsProps = path.join(path.dirname(slnFile), 'BuildFlags.props');
+  let changesNeeded = false;
+  if (fs.existsSync(buildFlagsProps)) {
+    const buildFlagsContents = configUtils.readProjectFile(buildFlagsProps);
+    const useWinUI3FromConfig = windowsProjectConfig.useWinUI3;
+    const useWinUI3FromBuildFlags =
+      configUtils
+        .tryFindPropertyValue(buildFlagsContents, 'UseWinUI3')
+        ?.toLowerCase() === 'true';
+    // use the UseWinUI3 value in react-native.config.js, or if not present, the value from BuildFlags.props
+    changesNeeded = fixPackagesConfig(
+      useWinUI3FromConfig !== undefined
+        ? useWinUI3FromConfig
+        : useWinUI3FromBuildFlags,
+      windowsProjectConfig,
+      checkMode,
+    );
+    if (useWinUI3FromConfig !== undefined) {
+      // Make sure BuildFlags matches the value that comes from react-native.config.js
+      const node = buildFlagsContents.getElementsByTagName('UseWinUI3');
+      const newValue = useWinUI3FromConfig ? 'true' : 'false';
+      changesNeeded = changesNeeded || node.item(0)?.textContent !== newValue;
+      if (!checkMode && changesNeeded) {
+        node.item(0)!.textContent = newValue;
+        const buildFlagsOutput = new XMLSerializer().serializeToString(
+          buildFlagsContents,
+        );
+        fs.writeFileSync(buildFlagsProps, buildFlagsOutput);
+      }
+    }
+  }
+  return changesNeeded;
+}
+
+function fixPackagesConfig(
```

Comment:
Can this method be rename specifically to something about selecting the right Xaml version?